### PR TITLE
enable validation of metadata form fields

### PR DESCRIPTION
**Pull request by akohlrusch**
Validation of the metadataform fields for the currently added layer. Hereafter, clean up the extisting form fields before uploading the next layer.

Please review

### CODE REVIEW

File: app/view/tab/CreateOrEditLayerController.js [hunk 1]
```diff
@@ -23,6 +23,10 @@ Ext.define('MoMo.admin.view.tab.CreateOrEditLayerController', {
 
         viewModel.set('layer', cleanLayer);
         viewModel.get('layer').set('id', undefined);
+
+        // ATTENTION !!!
+        // We destroy the view in the onHide method for cleanup reasons.
+        me.getView().destroy();
```
Comment (marcjansen) [inline app/view/tab/CreateOrEditLayerController.js:29]: Thabks for the explicit comment. We should revisit this later.

File: classic/src/view/panel/layer/General.js [hunk 2]
```diff
@@ -54,6 +54,28 @@ Ext.define('MoMo.admin.view.panel.layer.General',{
                 bind: {
                     hidden: '{!isNewLayer}'
                 },
+                msgTarget: 'under',
+                validator: function(val) {
+                    var startsWith = val.startsWith(' ');
```
Comment (marcjansen) [inline classic/src/view/panel/layer/General.js]: I fear that no all current browsers understand `startsWith`.

File: classic/src/view/panel/layer/Metadata.js [hunk 3]
```diff
@@ -12,317 +12,485 @@ Ext.define('MoMo.admin.view.panel.layer.Metadata',{
...
+                        else if( startsWithWhiteSpace && ! containsTwoWhitespaces && ! startsWithEmptyRow && ! containsTowEmptyRows) {
+                            return startsWithWhiteSpaceErrMsg + ' ' + containsTwoEmptyRowsErrMsg;
+                        }
+                        else if( startsWithWhiteSpace && containsTwoWhitespaces && ! startsWithEmptyRow && ! containsTowEmptyRows) {
+                            return startsWithWhiteSpaceErrMsg + ' '+ containsTwoWhitespacesErrMsg;
+                        }
+                        else if( startsWithWhiteSpace &&  containsTwoWhitespaces && ! startsWithEmptyRow && containsTowEmptyRows) {
+                            return startsWithWhiteSpaceErrMsg + ' '+ containsTwoWhitespacesErrMsg + ' '+ containsTowEmptyRowsErrMsg;
+                        }
+                    },*/
```
Comment (marcjansen) [inline classic/src/view/panel/layer/Metadata.js]: Please remove the commented code.

File: classic/src/view/panel/layer/Metadata.js [hunk 4]
```diff
@@ -12,317 +12,485 @@ Ext.define('MoMo.admin.view.panel.layer.Metadata',{
...
                         width: '100%'
                     },
                     items: [{
                         xtype: 'textfield',
-                        name: 'metadata-organisation-address-deliveryPoint',
+                        name: 'metadata-organisation-name',
                         allowBlank: false,
+                        msgTarget: 'under',
+                        validator: function(val) {
+                            var startsWith = val.startsWith(' ');
```
Comment (marcjansen) [inline classic/src/view/panel/layer/Metadata.js]: See above wrt `String.prototype.startsWith`

File: classic/src/view/panel/layer/Metadata.js [hunk 5]
```diff
@@ -12,317 +12,485 @@ Ext.define('MoMo.admin.view.panel.layer.Metadata',{
...
+                        defaults: {
+                            width: '100%'
+                        },
+                        items: [{
+                            xtype: 'textfield',
+                            name: 'metadata-organisation-address-deliveryPoint',
+                            allowBlank: false,
+                            msgTarget: 'under',
+                            validator: function(val) {
+                                var startsWith = val.startsWith(' ');
```
Comment (marcjansen) [inline classic/src/view/panel/layer/Metadata.js]: Same here. Can you try to create a (static) utility method? You can do this in a follow up PR.

File: classic/src/view/tab/CreateOrEditLayer.js [hunk 6]
```diff
@@ -58,6 +58,7 @@ Ext.define('MoMo.admin.view.tab.CreateOrEditLayer', {
     listeners: {
         afterrender: 'loadLayerData',
         show: 'loadLayerData',
+        // We destroy the view in the onHide method for cleanup reasons.
```
Comment (marcjansen) [inline classic/src/view/tab/CreateOrEditLayer.js]: Shudder. please add a TODO, so we may revisit it.

**Review by marcjansen — APPROVED** [review]
Thanks for tackling this. I left a few inline comments. Once they are addressed, this is good to go.

Good work!

File: classic/src/view/panel/layer/General.js (hunk 2 above)
Comment (akohlrusch) [reply classic/src/view/panel/layer/General.js]: The validator function has been moved in a new util class `MoMo.admin.util.TextfieldValidator`. Here `Ext.String.startsWith()` ist used. See 96e4936

File: classic/src/view/panel/layer/Metadata.js (hunk 3 above)
Comment (akohlrusch) [reply classic/src/view/panel/layer/Metadata.js]: Done.

File: classic/src/view/panel/layer/Metadata.js (hunk 4 above)
Comment (akohlrusch) [reply classic/src/view/panel/layer/Metadata.js]: The validator function has been moved in a new util class MoMo.admin.util.TextfieldValidator. Here Ext.String.startsWith() ist used. See 96e4936

File: classic/src/view/panel/layer/Metadata.js (hunk 5 above)
Comment (akohlrusch) [reply classic/src/view/panel/layer/Metadata.js]: The validator function has been moved in a new util class MoMo.admin.util.TextfieldValidator. Here Ext.String.startsWith() ist used. See 96e4936

File: classic/src/view/tab/CreateOrEditLayer.js (hunk 6 above)
Comment (akohlrusch) [reply classic/src/view/tab/CreateOrEditLayer.js]: Done. See 6a8092c.

**Comment by akohlrusch** [comment]
Have you more wishes marcjansen?

**Comment by marcjansen** [comment]
> Have you more wishes marcjansen?

Nop. I am as happy as can be.

Thanks and please merge

**Review by marcjansen — APPROVED** [review]

**Pull request merged**